source/extensions/filters/network/redis_proxy/conn_pool_impl.h

```diff
@@ -177,6 +179,7 @@ class InstanceImpl : public Instance, public std::enable_shared_from_this<Instan
     Upstream::ClusterUpdateCallbacksHandlePtr cluster_update_handle_;
     Upstream::ThreadLocalCluster* cluster_{};
     absl::node_hash_map<Upstream::HostConstSharedPtr, ThreadLocalActiveClientPtr> client_map_;
+    absl::node_hash_map<Upstream::HostConstSharedPtr, TokenBucketPtr> cx_rate_limiter_map_;
```

Comment:
Q: why isn't the token bucket part of `ThreadLocalActiveClient` (and hence we could just use `client_map_` instead)?

Reply:
The token bucket is used to control the initialization of `ThreadLocalActiveClient` so it is easier and more intuitive to have it outside of `ThreadLocalActiveClient`

Reply:
Fair point.